src/main/java/io/nats/client/Options.java

```diff
@@ -1710,6 +1710,10 @@ else if (useDefaultTls) {
                 }
             }
 
+            if (tlsFirst && sslContext == null) {
+                throw new IllegalStateException("SSL context required for tls handshake first");
+            }
+

```

Comment:
I did not used to check this before, but if there is no SSL Context, we can't do any tls anyway, so it's a fast fail essentially.